src/logics/Logic.cc

```diff
@@ -597,6 +586,9 @@ void Logic::simplifyTree(PTRef tr, PTRef& root_out)
 #endif
             for (int j = 0; j < t.size(); j++) {
                 PTRef cr = t[j];
+                if (isNonBoolIte(cr)) {
```

Comment:
Now that we have established that both boolean and non-boolean ITEs are treated the same way, this check seems suspicious. I think the processing should not stop also at boolean ITEs, also they are treated as variables now.

Reply:
Yes.  Just to be sure, do we agree it's the other way round, so the processing should stop at all ITEs.

Reply:
Yes, you are correct, I don't know why I wrote "not" there.